fsspec/implementations/dropboxdrivefs.py

```diff
@@ -0,0 +1,239 @@
+
+        if detail:
+            for item in list_item.entries:
+                if isinstance(item, dropbox.files.FileMetadata):
+                    list_file.append(
+                        {"name": item.path_display, "size": item.size, "type": "file"}
+                    )
+                elif isinstance(item, dropbox.files.FolderMetadata):
+                    list_file.append(
+                        {"name": item.path_display, "size": None, "type": "folder"}
+                    )
+                else:
+                    list_file.append(
+                        {"name": item.path_display, "size": item.size, "type": "unknow"}
+                    )
+        else:
+            for item in list_item.entries:
+                list_file.append(item.path_display)
+        return list_file
+
+    def _open(
+        self,
+        path,
+        mode="rb",
+        block_size=None,
+        autocommit=True,
+        cache_options=None,
+        **kwargs
+    ):
+        return DropboxDriveFile(
+            self,
+            path,
+            mode=mode,
+            block_size=4 * 1024 * 1024,
+            autocommit=autocommit,
+            cache_options=cache_options,
+            **kwargs
+        )
+
+    def info(self, url, **kwargs):
+        """Get info of URL
+        """
+        metadata = self.dbx.files_get_metadata(url)
+        if isinstance(metadata, dropbox.files.FileMetadata):
+            return {
+                "name": metadata.path_display,
+                "size": metadata.size,
+                "type": "file",
+            }
+        elif isinstance(metadata, dropbox.files.FolderMetadata):
+            return {"name": metadata.path_display, "size": None, "type": "folder"}
+        else:
+            return {"name": url, "size": None, "type": "unknow"}
+
+
+class DropboxDriveFile(AbstractBufferedFile):
+    """ fetch_all, fetch_range, and read method are based from the http implementation
+    """
+
+    def __init__(
+        self,
+        fs,
+        path,
+        mode="rb",
+        block_size="default",
+        autocommit=True,
+        cache_type="readahead",
+        cache_options=None,
+        **kwargs
+    ):
+        """
+        Open a file.
+        Parameters
+        ----------
+        fs: instance of DropboxDriveFileSystem
+        path : str
+            file path to inspect in dropbox
+        mode: str
+            Normal file modes.'rb' or 'wb'
+        block_size: int or None
+            The amount of read-ahead to do, in bytes. Default is 5MB, or the value
+            configured for the FileSystem creating this file
+        """
+        for key, value in kwargs.items():
+            print("{0} = {1}".format(key, value))
+        self.session = fs.session if fs.session is not None else requests.Session()
+        super().__init__(fs=fs, path=path, mode=mode, block_size=block_size, **kwargs)
+
+        self.path = path
+        self.dbx = self.fs.dbx
+        while "//" in path:
```

Comment:
`while` is not needed here, `replace` does the job in one go, unless you expect "//////"-like strings